x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -388,5 +401,12 @@ public int read(byte[] bytes, int off, int len) throws IOException {
 
             return bytesReceived;
         }
+
+        private void maybePause(int bytesRequested) {
+            long throttleTimeInNanos = rateLimiter.maybePause(bytesRequested);
+            if (throttleTimeInNanos > 0) {
```

Comment:
we can inline this method if we change rateLimiter.maybePause to always return >= 0.